src/cryptoadvance/specterext/stacktrack/controller.py

```diff
@@ -44,19 +47,67 @@ def index():
     return render_template("stacktrack/index.jinja")
 
 
-# @stacktrack_endpoint.route("/wallet_chart", methods=["GET"])
+@stacktrack_endpoint.route("/wallets_overview")
+@login_required
+def wallets_overview():
+    span: str = request.args.get("span")
+    span = "1y" if span is None else span
+
+    specter().check_blockheight()
+
+    wallets_overview_vm_dict = specter().service_manager.execute_ext_callbacks(
+        adjust_view_model, WalletsOverviewVm()
+    )
+    if len(wallets_overview_vm_dict) == 0:
+        wallets_overview_vm = WalletsOverviewVm()
+    elif len(wallets_overview_vm_dict.values()) == 1:
+        wallets_overview_vm = list(wallets_overview_vm_dict.values())[0]
+    else:
+        raise logger.error("Seems that we have more than one WalletsOverviewVm Extension")
+
+    wallets: list[Wallet] = list(specter().wallet_manager.wallets.values())
+    for wallet in wallets:
+        wallet.update_balance()
+        wallet.check_utxo()
+    txs: list = _extract_txs(wallets)
+    chart: go.Figure = getattr(plot, f"build_chart_{span}")(txs)
+
+    return render_template(
+        "wallet/overview/wallets_overview.jinja",
+        specter=specter(),
+        rand=rand,
+        services=specter().service_manager.services,
+        wallets_overview_vm=wallets_overview_vm,
+        active_span=span,
+        chart=chart,
+        url_path="wallets_overview",
+    )
+
+    return render_template("stacktrack/wallet/overview/wallets_overview.jinja")
+
+
 @stacktrack_endpoint.route("/wallet/<wallet_alias>/chart", methods=["GET"])
 @login_required
 def stacktrack_wallet_chart(wallet_alias: str) -> str:
-    span = request.args.get("span")
+    span: str = request.args.get("span")
     span = "1y" if span is None else span
-    wallet = app.specter.wallet_manager.get_by_alias(wallet_alias)
-    balance_plot: go.Figure = getattr(pbuild, f"build_plot_{span}")(wallet)
+    wallet: Wallet = app.specter.wallet_manager.get_by_alias(wallet_alias)
+    chart: go.Figure = getattr(plot, f"build_chart_{span}")(wallet.txlist())
```

Comment:
I don't understand why you use getattr here?

Reply:
It may be Python noobishness on my part. Basically I'm using reflection to pick out and invoke the desired `build_chart_{span}` function. I could do it with a bunch of if/elses but this seemed cleaner. Let me know if there's a better way to do this.

Reply:
Ok, can you refactor that into a function within the plot module? That way you just need reflection within the plot module and can encapsulate a bit better?

Reply:
Good idea. Will do.

Reply:
Great suggestion. Done.